### ISSUE TITLE
Thread Loop "Checking all queues begining with prefix"

### ISSUE THREAD

**Issue by sashokbg** [awslabs/amazon-sqs-java-temporary-queues-client#33]
I am trying to put temporary queues in place for bi-directional send and receive scenario where the producer is implemented using your library and the consumer is written manually in NodeJS since my lambda is based on JS.

Unfortunately though I have lots of problems with looping events like "Checking all queues begining with prefix" which seems to generate messages in the temporary queue which is really bad assuming it all costs money !

![sqs-messages](https://user-images.githubusercontent.com/10643824/74750124-84ba4680-526b-11ea-96e6-dc77b0dc86f8.gif)

The console messages I receive in the producer: 

```
2020-02-18 16:32:10.650 DEBUG 18099 --- [nsumer-Thread-1] c.a.s.sqs.MessageMD5ChecksumHandler      : Message body: .rO0ABXA=
2020-02-18 16:32:10.650 DEBUG 18099 --- [onPool-worker-5] c.a.s.sqs.MessageMD5ChecksumHandler      : Message body: .rO0ABXA=
2020-02-18 16:32:10.651 DEBUG 18099 --- [nsumer-Thread-1] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message body: 98e4c00964e43fd75ad2e2fa2f190887
2020-02-18 16:32:10.651 DEBUG 18099 --- [onPool-worker-5] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message body: 98e4c00964e43fd75ad2e2fa2f190887
2020-02-18 16:32:10.651 DEBUG 18099 --- [nsumer-Thread-1] c.a.s.sqs.MessageMD5ChecksumHandler      : Message attribtues: {__AmazonSQSVirtualQueuesClient.QueueName={StringValue: __RequesterClientQueues__cf226b82-ab67-4da2-9ac7-f3c372fe4baa,StringListValues: [],BinaryListValues: [],DataType: String}}
2020-02-18 16:32:10.651 DEBUG 18099 --- [onPool-worker-5] c.a.s.sqs.MessageMD5ChecksumHandler      : Message attribtues: {__AmazonSQSVirtualQueuesClient.QueueName={StringValue: __RequesterClientQueues__66bb67da-1e84-43f8-a1a8-c3c1852dd6e5,StringListValues: [],BinaryListValues: [],DataType: String}}
2020-02-18 16:32:10.651 DEBUG 18099 --- [nsumer-Thread-1] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message attributes: 3058cbdc6583f41873b1d3e14350f9a4
2020-02-18 16:32:10.651 DEBUG 18099 --- [onPool-worker-5] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message attributes: ccfa3e7753bb2b5e774a2b621bee72c4
2020-02-18 16:32:10.661 DEBUG 18099 --- [onPool-worker-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Message body: .rO0ABXA=
2020-02-18 16:32:10.661 DEBUG 18099 --- [onPool-worker-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message body: 98e4c00964e43fd75ad2e2fa2f190887
2020-02-18 16:32:10.661 DEBUG 18099 --- [onPool-worker-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Message attribtues: {__AmazonSQSVirtualQueuesClient.QueueName={StringValue: __RequesterClientQueues__931460ec-acfa-4cbd-95e4-faf482095a97,StringListValues: [],BinaryListValues: [],DataType: String}}
2020-02-18 16:32:10.661 DEBUG 18099 --- [onPool-worker-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message attributes: c1cb0df2c89955f9454b08d1b9258f38
2020-02-18 16:32:10.693 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Message body: .rO0ABXA=
2020-02-18 16:32:10.693 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message body: 98e4c00964e43fd75ad2e2fa2f190887
2020-02-18 16:32:10.693 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Message attribtues: {__AmazonSQSVirtualQueuesClient.QueueName={StringValue: __RequesterClientQueues__cf226b82-ab67-4da2-9ac7-f3c372fe4baa,StringListValues: [],BinaryListValues: [],DataType: String}}
2020-02-18 16:32:10.694 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message attributes: 3058cbdc6583f41873b1d3e14350f9a4
2020-02-18 16:32:10.694 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Message body: .rO0ABXA=
2020-02-18 16:32:10.694 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message body: 98e4c00964e43fd75ad2e2fa2f190887
2020-02-18 16:32:10.694 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Message attribtues: {__AmazonSQSVirtualQueuesClient.QueueName={StringValue: __RequesterClientQueues__66bb67da-1e84-43f8-a1a8-c3c1852dd6e5,StringListValues: [],BinaryListValues: [],DataType: String}}
2020-02-18 16:32:10.694 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message attributes: ccfa3e7753bb2b5e774a2b621bee72c4
2020-02-18 16:32:10.694 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Message body: .rO0ABXA=
2020-02-18 16:32:10.694 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message body: 98e4c00964e43fd75ad2e2fa2f190887
2020-02-18 16:32:10.694 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Message attribtues: {__AmazonSQSVirtualQueuesClient.QueueName={StringValue: __RequesterClientQueues__931460ec-acfa-4cbd-95e4-faf482095a97,StringListValues: [],BinaryListValues: [],DataType: String}}
2020-02-18 16:32:10.694 DEBUG 18099 --- [nsumer-Thread-3] c.a.s.sqs.MessageMD5ChecksumHandler      : Expected  MD5 of message attributes: c1cb0df2c89955f9454b08d1b9258f38
2020-02-18 16:32:10.699 DEBUG 18099 --- [nsumer-Thread-7] c.a.s.sqs.AmazonSQSVirtualQueuesClient   : Deleting Virtual Queue is 4 and Queue Name is https://sqs.eu-west-1.amazonaws.com/122333224379/__RequesterClientQueues__8e0145ad-8f87-4c15-8d55-733a96f950b2-0#__RequesterClientQueues__cf226b82-ab67-4da2-9ac7-f3c372fe4baa
2020-02-18 16:32:10.699 DEBUG 18099 --- [nsumer-Thread-5] c.a.s.sqs.AmazonSQSVirtualQueuesClient   : Deleting Virtual Queue is 4 and Queue Name is https://sqs.eu-west-1.amazonaws.com/122333224379/__RequesterClientQueues__8e0145ad-8f87-4c15-8d55-733a96f950b2-0#__RequesterClientQueues__931460ec-acfa-4cbd-95e4-faf482095a97
2020-02-18 16:32:10.699 DEBUG 18099 --- [nsumer-Thread-8] c.a.s.sqs.AmazonSQSVirtualQueuesClient   : Deleting Virtual Queue is 4 and Queue Name is https://sqs.eu-west-1.amazonaws.com/122333224379/__RequesterClientQueues__8e0145ad-8f87-4c15-8d55-733a96f950b2-0#__RequesterClientQueues__66bb67da-1e84-43f8-a1a8-c3c1852dd6e5


... skipped hundreds of more ...

```

Here is the producer code:

```
      try {
        com.amazonaws.services.sqs.model.SendMessageRequest request = new com.amazonaws.services.sqs.model.SendMessageRequest()
            .withMessageBody(objectMapper.writeValueAsString(domainEvent))
            .withQueueUrl(topic.getTopicURL());

        Message message = sqsRequester.sendMessageAndGetResponse(request, 20, TimeUnit.SECONDS);

        result.complete(objectMapper.readValue(message.getBody(), responseType));
      } catch (JsonProcessingException | TimeoutException e) {
        log.error("An error has occurred publishing domain event to SQS: {}", domainEvent, e);
      }
```

And the node JS consumer code:
If I understand correctly the consumer needs to answer to the request on the provided queue name for the response.

```
function sendSuccess(callback, response, queueUrl) {
  let params = {
    MessageBody: JSON.stringify(response),
    QueueUrl: queueUrl,
    MessageAttributes: {
      "__AmazonSQSVirtualQueuesClient.QueueName": {
        DataType: 'String',
        StringValue: queueUrl.split('#')[1]
      }
    }
  };

  sqs.sendMessage(params, function(sqsErr) {
    if(sqsErr) {
      console.log(`Problem answering to SQS event`, sqsErr);
      callback(sqsErr, response);
    } else {
      callback(null, response);
    }
  });
}
```

Please advise.
Best Regards
Alexander KIRILOV

**Comment by robin-aws** [comment]
Hello @sashokbg!

What version of the library are you using? Those "Deleting Virtual Queue" log messages were downgraded in severity as of v1.0.1, so I'd recommend you pick up the latest release if you haven't already.

If you are getting excessive "Checking all queues begining with prefix" messages, you might be running afoul of #28, which there is a PR out to fix: #32. If so I will prioritize getting that reviewed and merged.

**Comment by sashokbg** [comment]
Hello @robin-aws yes I am using the latest version - 1.0.2 and still get the verbose messages.
The main problem though is not verbosity but my concern that it actually creates lots of SQS messages as you can see from my GIF - on every refresh I see message numbers that change which means create / delete message operation.

As it seems it may be related to #28. I will do some debugging and if it is the case we can mark as duplicate.

**Comment by sashokbg** [comment]
Duplicate of #28

**Comment by SStorm** [comment]
@robin-aws is there an ETA for a new release? I can see #32 is already merged, so would be good to get 1.0.3 that fixes the issue. Without it the library is sadly unusable.

**Comment by robin-aws** [comment]
Hi @SStorm! I'm going to work with the SQS team next week to get a release prioritized - no ETA yet but I will update as soon as we have one.

**Comment by khoa-tran** [comment]
@robin-aws any update on the next release?

**Comment by SStorm** [comment]
@robin-aws ping, new release please?

**Comment by adam-aws** [comment]
Closing this since #28 was released in v1.1.0
